pkg/apis/servicecatalog/v1beta1/types.go

```diff
@@ -865,6 +865,10 @@ type ServiceInstanceSpec struct {
 	// been made to the secrets from which the parameters are sourced.
 	// +optional
 	UpdateRequests int64 `json:"updateRequests"`
+
+	// DeletionRequested provides an imperative means to request
+	// deletion by doing an update.
+	DeletionRequested bool `json:"deletionRequested"`
```

Comment:
I do not think this matches with the rest of kubernetes nor does it mesh with the norms of OSB. I would suggest:

`deprovisionTimestamp time.Time`

Then you can set the time the deprovision was requested. That is declarative.